x/evm/types/msg_retry_failed_event.go

```diff
@@ -0,0 +1,55 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/axelarnetwork/axelar-core/utils"
+)
+
+// NewRetryFailedEventRequest - RetryFailedEventRequest constructor
+func NewRetryFailedEventRequest(sender sdk.AccAddress, chain string, eventID string) *RetryFailedEventRequest {
+	return &RetryFailedEventRequest{
+		Sender:  sender,
+		Chain:   utils.NormalizeString(chain),
+		EventID: utils.NormalizeString(eventID),
+	}
+}
+
+// Route returns the route for this message
+func (m RetryFailedEventRequest) Route() string {
+	return RouterKey
+}
+
+// Type returns the type of the message
+func (m RetryFailedEventRequest) Type() string {
+	return "RetryFailedEvent"
+}
+
+// ValidateBasic executes a stateless message validation
+func (m RetryFailedEventRequest) ValidateBasic() error {
+	if err := sdk.VerifyAddressFormat(m.Sender); err != nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, sdkerrors.Wrap(err, "sender").Error())
+	}
+
+	if err := utils.ValidateString(m.Chain); err != nil {
+		return sdkerrors.Wrap(err, "invalid chain")
+	}
+
+	if err := utils.ValidateString(m.EventID); err != nil {
```

Comment:
`.EventID` should be validated even further, making sure that it's of the format `${txID}-${index}`

Reply:
maybe even create a type for event ID.

Reply:
good idea